crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -493,7 +493,7 @@ impl<'a> Checker<'a> {
     /// Push `diagnostic` if the checker is not in a `@no_type_check` context.
     pub(crate) fn push_type_diagnostic(&mut self, diagnostic: Diagnostic) {
```

Comment:
We may want to rename this method to `report_type_diagnostic` and also change its signature to take a `&self` (unless you've already done so in a higher up commit)